fontbe/src/kern.rs

```diff
@@ -9,55 +9,91 @@ use fea_rs::{
 use fontdrasil::{
     coords::NormalizedLocation,
     orchestration::{Access, AccessBuilder, Work},
+    types::GlyphName,
 };
 use fontir::{
-    ir::{KernPair, KernParticipant},
+    ir::{GlyphOrder, KernPair, KernParticipant},
     orchestration::WorkId as FeWorkId,
 };
-use ordered_float::OrderedFloat;
+use log::debug;
 use write_fonts::types::GlyphId;
 
 use crate::{
     error::Error,
     features::resolve_variable_metric,
-    orchestration::{AnyWorkId, BeWork, Context, Kerning, WorkId},
+    orchestration::{
+        AnyWorkId, BeWork, Context, KernAdjustments, KernSegment, Kerning, Kerns, PairPosEntry,
+        WorkId,
+    },
 };
 
+/// On Linux it took ~0.01 ms per loop, try to get enough to make fan out worthwhile
+/// based on empirical testing
+const KERNS_PER_BLOCK: usize = 2048;
+
+#[derive(Debug)]
+struct KerningWork;
+
+#[derive(Debug)]
+struct KerningSegmentWork {
+    segment: usize,
+}
+
 #[derive(Debug)]
-struct KerningWork {}
+struct KerningGatherWork;
 
 pub fn create_kerning_work() -> Box<BeWork> {
     Box::new(KerningWork {})
 }
 
+pub fn create_kern_segment_work(kern_pairs: &Kerning) -> Vec<Box<BeWork>> {
+    let segments = kern_pairs.adjustments.len().div_ceil(KERNS_PER_BLOCK);
+    let mut work: Vec<Box<BeWork>> = Vec::with_capacity(segments);
+    debug!(
+        "Process {} kerning adjustments in {} chunks",
+        kern_pairs.adjustments.len(),
+        segments
+    );
+    for segment in 0..segments {
+        work.push(Box::new(KerningSegmentWork { segment }));
+    }
+    work
+}
+
+pub fn create_kerns_work() -> Box<BeWork> {
+    Box::new(KerningGatherWork {})
+}
+
+fn gid(glyph_order: &GlyphOrder, name: &GlyphName) -> Result<GlyphId, Error> {
+    glyph_order
+        .glyph_id(name)
+        .map(|gid| GlyphId::new(gid as u16))
+        .ok_or_else(|| Error::MissingGlyphId(name.clone()))
+}
```

Comment:
Maybe we just make this a method on `GlyphOrder`? Not sure why I didn't do that initially, probably no good reason.

Reply:
Ha, indeed. Will do.